File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.swift
class Solution {
    func maxProfit(_ a: [Int]) -> Int {
        let n=a.count
        var Min=a[0]
        var prof=0
        for i in 1..<n {
            let cost=a[i]-Min
            prof=max(prof,cost)
            Min=min(Min,a[i])
        }
        return prof
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int Min=prices[0];
        int n=prices.size();
        int prof=0;
        for(int i=1;i<n;i++){
            int cost=prices[i]-Min;
            prof=max(prof,cost);
            Min=min(Min,prices[i]);
        }
        return prof;
    }
};